<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.313">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>16S Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="10_16S_Analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="10_16S_Analysis_files/libs/quarto-html/quarto.js"></script>
<script src="10_16S_Analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="10_16S_Analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="10_16S_Analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="10_16S_Analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="10_16S_Analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="10_16S_Analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="10_16S_Analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="10_16S_Analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="10_16S_Analysis_files/libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<script src="10_16S_Analysis_files/libs/plotly-binding-4.10.3/plotly.js"></script>
<script src="10_16S_Analysis_files/libs/typedarray-0.1/typedarray.min.js"></script>
<script src="10_16S_Analysis_files/libs/jquery-3.5.1/jquery.min.js"></script>
<link href="10_16S_Analysis_files/libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="10_16S_Analysis_files/libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="10_16S_Analysis_files/libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="10_16S_Analysis_files/libs/plotly-main-2.11.1/plotly-latest.min.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup">Setup</a>
  <ul class="collapse">
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link" data-scroll-target="#load-packages">Load packages</a></li>
  <li><a href="#load-in-16s-data" id="toc-load-in-16s-data" class="nav-link" data-scroll-target="#load-in-16s-data">Load in 16S data</a></li>
  <li><a href="#construct-metadata-files" id="toc-construct-metadata-files" class="nav-link" data-scroll-target="#construct-metadata-files">Construct metadata files</a></li>
  <li><a href="#construct-phyloseq-components" id="toc-construct-phyloseq-components" class="nav-link" data-scroll-target="#construct-phyloseq-components">Construct phyloseq components</a></li>
  <li><a href="#construct-phyloseq-object-of-all-samples" id="toc-construct-phyloseq-object-of-all-samples" class="nav-link" data-scroll-target="#construct-phyloseq-object-of-all-samples">Construct phyloseq object of all samples</a></li>
  </ul></li>
  <li><a href="#decontam-ination" id="toc-decontam-ination" class="nav-link" data-scroll-target="#decontam-ination">Decontam-ination</a>
  <ul class="collapse">
  <li><a href="#inspect-library-sizes" id="toc-inspect-library-sizes" class="nav-link" data-scroll-target="#inspect-library-sizes">Inspect library sizes</a></li>
  <li><a href="#identify-contaminants-prevalence" id="toc-identify-contaminants-prevalence" class="nav-link" data-scroll-target="#identify-contaminants-prevalence">Identify contaminants: prevalence</a></li>
  <li><a href="#remove-contaminants-from-our-ps-object" id="toc-remove-contaminants-from-our-ps-object" class="nav-link" data-scroll-target="#remove-contaminants-from-our-ps-object">Remove contaminants from our PS object</a></li>
  <li><a href="#validate-and-finalize-our-ps-object" id="toc-validate-and-finalize-our-ps-object" class="nav-link" data-scroll-target="#validate-and-finalize-our-ps-object">Validate and finalize our PS object</a></li>
  </ul></li>
  <li><a href="#cache" id="toc-cache" class="nav-link" data-scroll-target="#cache">Cache</a></li>
  <li><a href="#analysis---all-samples" id="toc-analysis---all-samples" class="nav-link" data-scroll-target="#analysis---all-samples">Analysis - All Samples</a>
  <ul class="collapse">
  <li><a href="#alpha-diversity-simpsons" id="toc-alpha-diversity-simpsons" class="nav-link" data-scroll-target="#alpha-diversity-simpsons">Alpha Diversity (Simpson’s)</a></li>
  <li><a href="#ordination" id="toc-ordination" class="nav-link" data-scroll-target="#ordination">Ordination</a></li>
  </ul></li>
  <li><a href="#analysis---1-field-samples" id="toc-analysis---1-field-samples" class="nav-link" data-scroll-target="#analysis---1-field-samples">Analysis - (1) Field Samples</a>
  <ul class="collapse">
  <li><a href="#alpha-diversity-simpsons-1" id="toc-alpha-diversity-simpsons-1" class="nav-link" data-scroll-target="#alpha-diversity-simpsons-1">Alpha Diversity (Simpson’s)</a></li>
  <li><a href="#composition" id="toc-composition" class="nav-link" data-scroll-target="#composition">Composition</a></li>
  </ul></li>
  <li><a href="#analysis---2-13c-experiments-tac" id="toc-analysis---2-13c-experiments-tac" class="nav-link" data-scroll-target="#analysis---2-13c-experiments-tac">Analysis - (2) 13C Experiments (TAC)</a>
  <ul class="collapse">
  <li><a href="#alpha-diversity-simpsons-2" id="toc-alpha-diversity-simpsons-2" class="nav-link" data-scroll-target="#alpha-diversity-simpsons-2">Alpha Diversity (Simpson’s)</a></li>
  <li><a href="#composition-1" id="toc-composition-1" class="nav-link" data-scroll-target="#composition-1">Composition</a></li>
  </ul></li>
  <li><a href="#analysis---3-d2o-experiments" id="toc-analysis---3-d2o-experiments" class="nav-link" data-scroll-target="#analysis---3-d2o-experiments">Analysis - (3) D2O Experiments</a>
  <ul class="collapse">
  <li><a href="#composition-2" id="toc-composition-2" class="nav-link" data-scroll-target="#composition-2">Composition</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">16S Analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>From <code>09_16S_Data_Reduction.qmd</code>, we have output caches and text files:</p>
<p>Summary of output files:</p>
<ol type="1">
<li><code>seqtab_final.txt</code> - A tab-delimited sequence-by-sample (i.e.&nbsp;OTU) table</li>
<li><code>tax_final.txt</code> - a tab-demilimited file showing the relationship between ASVs, ASV IDs, and their taxonomy</li>
<li><code>seqtab_wTax_mctoolsr.txt</code> - a tab-delimited file with ASVs as rows, samples as columns and the final column showing the taxonomy of the ASV ID</li>
<li><code>repset.fasta</code> - a fasta file with the representative sequence of each ASV. Fasta headers are the ASV ID and taxonomy string.</li>
</ol>
<section id="setup" class="level2">
<h2 class="anchored" data-anchor-id="setup">Setup</h2>
<section id="load-packages" class="level3">
<h3 class="anchored" data-anchor-id="load-packages">Load packages</h3>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'lubridate' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.4     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors
Loading required package: BiocGenerics


Attaching package: 'BiocGenerics'


The following objects are masked from 'package:lubridate':

    intersect, setdiff, union


The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union


The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs


The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min


Loading required package: S4Vectors

Loading required package: stats4


Attaching package: 'S4Vectors'


The following objects are masked from 'package:lubridate':

    second, second&lt;-


The following objects are masked from 'package:dplyr':

    first, rename


The following object is masked from 'package:tidyr':

    expand


The following object is masked from 'package:utils':

    findMatches


The following objects are masked from 'package:base':

    expand.grid, I, unname


Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'IRanges' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'

The following object is masked from 'package:lubridate':

    %within%

The following objects are masked from 'package:dplyr':

    collapse, desc, slice

The following object is masked from 'package:purrr':

    reduce

The following object is masked from 'package:phyloseq':

    distance

Loading required package: XVector

Attaching package: 'XVector'

The following object is masked from 'package:purrr':

    compact

Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'GenomeInfoDb' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'

The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'microViz' was built under R version 4.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>microViz version 0.11.0 - Copyright (C) 2023 David Barnett
! Website: https://david-barnett.github.io/microViz
✔ Useful?  For citation details, run: `citation("microViz")`
✖ Silence? `suppressPackageStartupMessages(library(microViz))`</code></pre>
</div>
</div>
</section>
<section id="load-in-16s-data" class="level3">
<h3 class="anchored" data-anchor-id="load-in-16s-data">Load in 16S data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>taxtable <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="st">"data/16S/outputs/03_tabletax/tax_final.rds"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb10-2"><a href="#cb10-2"></a>  <span class="co"># convert to tibble</span></span>
<span id="cb10-3"><a href="#cb10-3"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span> </span>
<span id="cb10-4"><a href="#cb10-4"></a>  <span class="co"># fix column headers</span></span>
<span id="cb10-5"><a href="#cb10-5"></a>  <span class="fu">mutate</span>(<span class="at">ASV_id =</span> <span class="fu">paste</span>(<span class="st">"ASV"</span>, <span class="fu">row_number</span>(), <span class="at">sep =</span> <span class="st">"_"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"ASV_id"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="co"># convert back to a matrix</span></span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb10-9"><a href="#cb10-9"></a></span>
<span id="cb10-10"><a href="#cb10-10"></a>seq_tab <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"data/16S/outputs/03_tabletax/seqtab_final.txt"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb10-11"><a href="#cb10-11"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"...1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
Rows: 978 Columns: 67
── Column specification
──────────────────────────────────────────────────────── Delimiter: "\t" chr
(1): ...1 dbl (66): BA1B-150-ac-A, BA1B-150-bic-A, BA1B-150-ENV,
BA1B-150-form-A, BA1B...
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
</div>
</div>
</section>
<section id="construct-metadata-files" class="level3">
<h3 class="anchored" data-anchor-id="construct-metadata-files">Construct metadata files</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>metadata <span class="ot">&lt;-</span> seq_tab <span class="sc">|&gt;</span> <span class="fu">colnames</span>() <span class="sc">|&gt;</span> <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span> </span>
<span id="cb12-2"><a href="#cb12-2"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"sample_id"</span> <span class="ot">=</span> <span class="st">"value"</span>) <span class="sc">|&gt;</span></span>
<span id="cb12-3"><a href="#cb12-3"></a>  <span class="co"># parse well ID</span></span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="fu">mutate</span>(</span>
<span id="cb12-5"><a href="#cb12-5"></a>    <span class="at">well =</span> <span class="fu">case_when</span>(</span>
<span id="cb12-6"><a href="#cb12-6"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"BA1B"</span>) <span class="sc">~</span> <span class="st">"BA1B"</span>,</span>
<span id="cb12-7"><a href="#cb12-7"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"BA1D"</span>) <span class="sc">~</span> <span class="st">"BA1D"</span>,</span>
<span id="cb12-8"><a href="#cb12-8"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"BA3A"</span>) <span class="sc">~</span> <span class="st">"BA3A"</span>,</span>
<span id="cb12-9"><a href="#cb12-9"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"BA4A"</span>) <span class="sc">~</span> <span class="st">"BA4A"</span>,</span>
<span id="cb12-10"><a href="#cb12-10"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"BLANK"</span></span>
<span id="cb12-11"><a href="#cb12-11"></a>    ),</span>
<span id="cb12-12"><a href="#cb12-12"></a>    <span class="co"># is it a blank?</span></span>
<span id="cb12-13"><a href="#cb12-13"></a>    <span class="at">is_blank =</span> <span class="fu">if_else</span>(</span>
<span id="cb12-14"><a href="#cb12-14"></a>      well <span class="sc">==</span> <span class="st">"BLANK"</span>,</span>
<span id="cb12-15"><a href="#cb12-15"></a>      <span class="at">true =</span> <span class="cn">TRUE</span>,</span>
<span id="cb12-16"><a href="#cb12-16"></a>      <span class="at">false =</span> <span class="cn">FALSE</span></span>
<span id="cb12-17"><a href="#cb12-17"></a>    ),</span>
<span id="cb12-18"><a href="#cb12-18"></a>    <span class="co"># parse depth</span></span>
<span id="cb12-19"><a href="#cb12-19"></a>    <span class="at">depth =</span> <span class="fu">parse_number</span>(<span class="fu">str_extract</span>(sample_id, <span class="st">"</span><span class="sc">\\</span><span class="st">d{2,3}</span><span class="sc">\\</span><span class="st">D"</span>)),</span>
<span id="cb12-20"><a href="#cb12-20"></a>    <span class="at">depth =</span> <span class="fu">if_else</span>(</span>
<span id="cb12-21"><a href="#cb12-21"></a>      is_blank <span class="sc">==</span> <span class="cn">TRUE</span>,</span>
<span id="cb12-22"><a href="#cb12-22"></a>      <span class="at">true =</span> <span class="cn">NA</span>,</span>
<span id="cb12-23"><a href="#cb12-23"></a>      <span class="at">false =</span> depth</span>
<span id="cb12-24"><a href="#cb12-24"></a>    ),</span>
<span id="cb12-25"><a href="#cb12-25"></a>    <span class="co"># parse which experiment the sample is a part of</span></span>
<span id="cb12-26"><a href="#cb12-26"></a>    <span class="at">experiment =</span> <span class="fu">case_when</span>(</span>
<span id="cb12-27"><a href="#cb12-27"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"H2O"</span>) <span class="sc">~</span> <span class="st">"D2O-SIP"</span>,</span>
<span id="cb12-28"><a href="#cb12-28"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"D2O"</span>) <span class="sc">~</span> <span class="st">"D2O-SIP"</span>,</span>
<span id="cb12-29"><a href="#cb12-29"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"13C-SIP"</span></span>
<span id="cb12-30"><a href="#cb12-30"></a>    ),</span>
<span id="cb12-31"><a href="#cb12-31"></a>    <span class="at">amendment =</span> <span class="fu">case_when</span>(</span>
<span id="cb12-32"><a href="#cb12-32"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"ac"</span>) <span class="sc">~</span> <span class="st">"acetate"</span>,</span>
<span id="cb12-33"><a href="#cb12-33"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"bic"</span>) <span class="sc">~</span> <span class="st">"bicarbonate"</span>,</span>
<span id="cb12-34"><a href="#cb12-34"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"form"</span>) <span class="sc">~</span> <span class="st">"formate"</span>,</span>
<span id="cb12-35"><a href="#cb12-35"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"ENV"</span>) <span class="sc">~</span> <span class="st">"pre-incubation"</span>,</span>
<span id="cb12-36"><a href="#cb12-36"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"Acet"</span>) <span class="sc">~</span> <span class="st">"acetate"</span>,</span>
<span id="cb12-37"><a href="#cb12-37"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"Form"</span>) <span class="sc">~</span> <span class="st">"formate"</span>,</span>
<span id="cb12-38"><a href="#cb12-38"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"Bic"</span>) <span class="sc">~</span> <span class="st">"bicarbonate"</span>,</span>
<span id="cb12-39"><a href="#cb12-39"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"D2O"</span>) <span class="sc">~</span> <span class="st">"D2O"</span>,</span>
<span id="cb12-40"><a href="#cb12-40"></a>      <span class="fu">str_detect</span>(sample_id, <span class="st">"H2O"</span>) <span class="sc">~</span> <span class="cn">NA</span>,</span>
<span id="cb12-41"><a href="#cb12-41"></a>    )</span>
<span id="cb12-42"><a href="#cb12-42"></a>  ) <span class="sc">|&gt;</span> </span>
<span id="cb12-43"><a href="#cb12-43"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"sample_id"</span>)</span>
<span id="cb12-44"><a href="#cb12-44"></a></span>
<span id="cb12-45"><a href="#cb12-45"></a><span class="co"># write the metadata to file</span></span>
<span id="cb12-46"><a href="#cb12-46"></a>writexl<span class="sc">::</span><span class="fu">write_xlsx</span>(metadata, <span class="at">path =</span> <span class="st">"data/sample_metadata_16S.xlsx"</span>)</span>
<span id="cb12-47"><a href="#cb12-47"></a></span>
<span id="cb12-48"><a href="#cb12-48"></a></span>
<span id="cb12-49"><a href="#cb12-49"></a><span class="co"># Pull out the sample list in our ideal order:</span></span>
<span id="cb12-50"><a href="#cb12-50"></a>sample_list_ordered <span class="ot">&lt;-</span> metadata <span class="sc">|&gt;</span> </span>
<span id="cb12-51"><a href="#cb12-51"></a>  <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span> </span>
<span id="cb12-52"><a href="#cb12-52"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span> </span>
<span id="cb12-53"><a href="#cb12-53"></a>  <span class="fu">arrange</span>(well, depth, amendment, is_blank) <span class="sc">|&gt;</span> </span>
<span id="cb12-54"><a href="#cb12-54"></a>  <span class="fu">select</span>(rowname) <span class="sc">|&gt;</span></span>
<span id="cb12-55"><a href="#cb12-55"></a>  <span class="fu">pull</span>()</span>
<span id="cb12-56"><a href="#cb12-56"></a></span>
<span id="cb12-57"><a href="#cb12-57"></a>sample_list_ordered_field <span class="ot">&lt;-</span> metadata <span class="sc">|&gt;</span> </span>
<span id="cb12-58"><a href="#cb12-58"></a>  <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span> </span>
<span id="cb12-59"><a href="#cb12-59"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span> </span>
<span id="cb12-60"><a href="#cb12-60"></a>  <span class="fu">filter</span>(amendment <span class="sc">==</span> <span class="st">"pre-incubation"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb12-61"><a href="#cb12-61"></a>  <span class="fu">arrange</span>(well, depth, amendment, is_blank) <span class="sc">|&gt;</span> </span>
<span id="cb12-62"><a href="#cb12-62"></a>  <span class="fu">select</span>(rowname) <span class="sc">|&gt;</span></span>
<span id="cb12-63"><a href="#cb12-63"></a>  <span class="fu">pull</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="construct-phyloseq-components" class="level3">
<h3 class="anchored" data-anchor-id="construct-phyloseq-components">Construct phyloseq components</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># convert the seq_tab file to an 'otu_table' phyloseq component</span></span>
<span id="cb13-2"><a href="#cb13-2"></a>asv_table <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(seq_tab, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-3"><a href="#cb13-3"></a></span>
<span id="cb13-4"><a href="#cb13-4"></a><span class="co"># convert the taxtable file to a 'TaxonomyTable' phyloseq component</span></span>
<span id="cb13-5"><a href="#cb13-5"></a>taxonomy_table <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">tax_table</span>(taxtable)</span>
<span id="cb13-6"><a href="#cb13-6"></a></span>
<span id="cb13-7"><a href="#cb13-7"></a><span class="co"># convert metadata into a 'sample_data' phyloseq component</span></span>
<span id="cb13-8"><a href="#cb13-8"></a>sample_vars <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(metadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="construct-phyloseq-object-of-all-samples" class="level3">
<h3 class="anchored" data-anchor-id="construct-phyloseq-object-of-all-samples">Construct phyloseq object of all samples</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(asv_table, taxonomy_table, sample_vars)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="decontam-ination" class="level2">
<h2 class="anchored" data-anchor-id="decontam-ination">Decontam-ination</h2>
<p>Here we apply <code>decontam</code> package to remove contaminants detected in our negative controls.</p>
<section id="inspect-library-sizes" class="level3">
<h3 class="anchored" data-anchor-id="inspect-library-sizes">Inspect library sizes</h3>
<p>First, let’s take a look at library size (number of reads) in each sample as a funciton of whether that sample was a positive sample or a negative control.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="co"># Put sample_data into a ggplot-friendly data.frame</span></span>
<span id="cb15-2"><a href="#cb15-2"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">sample_data</span>(ps)) </span>
<span id="cb15-3"><a href="#cb15-3"></a></span>
<span id="cb15-4"><a href="#cb15-4"></a><span class="co"># returns total number of individuals observed from each sample</span></span>
<span id="cb15-5"><a href="#cb15-5"></a>df<span class="sc">$</span>LibrarySize <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(ps) </span>
<span id="cb15-6"><a href="#cb15-6"></a></span>
<span id="cb15-7"><a href="#cb15-7"></a>df <span class="ot">&lt;-</span> df[<span class="fu">order</span>(df<span class="sc">$</span>LibrarySize),]</span>
<span id="cb15-8"><a href="#cb15-8"></a>df<span class="sc">$</span>Index <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">nrow</span>(df)) <span class="co"># create a new column called Index that's a simple index val</span></span>
<span id="cb15-9"><a href="#cb15-9"></a></span>
<span id="cb15-10"><a href="#cb15-10"></a><span class="fu">ggplot</span>(<span class="at">data=</span>df, <span class="fu">aes</span>(<span class="at">x=</span>Index, <span class="at">y=</span>LibrarySize, <span class="at">color=</span>is_blank, <span class="at">label =</span> <span class="fu">rownames</span>(df))) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="1152"></p>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co"># and do a plotly!</span></span>
<span id="cb16-2"><a href="#cb16-2"></a>plotly<span class="sc">::</span><span class="fu">ggplotly</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-f8a79d8bd40b4ab008e2" style="width:100%;height:433px;"></div>
<script type="application/json" data-for="htmlwidget-f8a79d8bd40b4ab008e2">{"x":{"data":[{"x":[11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66],"y":[20499,28506,28633,28947,29301,29818,31006,31278,31667,31944,32204,32706,32751,33021,34109,34658,34676,34860,35267,35312,35354,35691,35849,36201,36367,37023,37497,38230,38245,38394,38432,38579,38618,38770,39312,39665,39750,39837,40200,41028,41253,41470,41497,41559,41921,42889,44081,44403,44949,45475,46017,46771,49213,51047,59435],"text":["Index: 11<br />LibrarySize: 20499<br />is_blank: FALSE<br />rownames(df): BA4A-270B-D2O-Acet","Index: 12<br />LibrarySize: 28506<br />is_blank: FALSE<br />rownames(df): BA1B-150-ac-A","Index: 13<br />LibrarySize: 28633<br />is_blank: FALSE<br />rownames(df): BA1B-20-bic-A","Index: 14<br />LibrarySize: 28947<br />is_blank: FALSE<br />rownames(df): BA4A-270A-D2O-Form","Index: 15<br />LibrarySize: 29301<br />is_blank: FALSE<br />rownames(df): BA1B-150-form-A","Index: 16<br />LibrarySize: 29818<br />is_blank: FALSE<br />rownames(df): BA4A-270B-D2O","Index: 17<br />LibrarySize: 31006<br />is_blank: FALSE<br />rownames(df): BA1B-20-form-A","Index: 18<br />LibrarySize: 31278<br />is_blank: FALSE<br />rownames(df): BA4A-150-bic-A","Index: 19<br />LibrarySize: 31667<br />is_blank: FALSE<br />rownames(df): BA1B-20-ENV","Index: 20<br />LibrarySize: 31944<br />is_blank: FALSE<br />rownames(df): BA1B-150-bic-A","Index: 21<br />LibrarySize: 32204<br />is_blank: FALSE<br />rownames(df): BA1B-250-ac-A","Index: 22<br />LibrarySize: 32706<br />is_blank: FALSE<br />rownames(df): BA3A-20-ENV","Index: 23<br />LibrarySize: 32751<br />is_blank: FALSE<br />rownames(df): BA1B-20-ac-A","Index: 24<br />LibrarySize: 33021<br />is_blank: FALSE<br />rownames(df): BA4A-150-ac-A","Index: 25<br />LibrarySize: 34109<br />is_blank: FALSE<br />rownames(df): BA1D-120B-D2O-Bic","Index: 26<br />LibrarySize: 34658<br />is_blank: FALSE<br />rownames(df): BA1B-250B-D2O-Form","Index: 27<br />LibrarySize: 34676<br />is_blank: FALSE<br />rownames(df): BA3A-150-ENV","Index: 28<br />LibrarySize: 34860<br />is_blank: FALSE<br />rownames(df): BA1B-250B-H2O","Index: 29<br />LibrarySize: 35267<br />is_blank: FALSE<br />rownames(df): BA1B-250B-D2O","Index: 30<br />LibrarySize: 35312<br />is_blank: FALSE<br />rownames(df): BA4A-20-ENV","Index: 31<br />LibrarySize: 35354<br />is_blank: FALSE<br />rownames(df): BA1B-250A-D2O","Index: 32<br />LibrarySize: 35691<br />is_blank: FALSE<br />rownames(df): BA1D-120B-D2O-Form","Index: 33<br />LibrarySize: 35849<br />is_blank: FALSE<br />rownames(df): BA4A-20-form-A","Index: 35<br />LibrarySize: 36201<br />is_blank: FALSE<br />rownames(df): BA4A-270B-D2O-Bic","Index: 36<br />LibrarySize: 36367<br />is_blank: FALSE<br />rownames(df): BA1B-250B-D2O-Acet","Index: 37<br />LibrarySize: 37023<br />is_blank: FALSE<br />rownames(df): BA4A-270-bic-A","Index: 38<br />LibrarySize: 37497<br />is_blank: FALSE<br />rownames(df): BA1B-250-form-A","Index: 39<br />LibrarySize: 38230<br />is_blank: FALSE<br />rownames(df): BA1D-120A-D2O-Bic","Index: 40<br />LibrarySize: 38245<br />is_blank: FALSE<br />rownames(df): BA1D-120A-H2O","Index: 41<br />LibrarySize: 38394<br />is_blank: FALSE<br />rownames(df): BA4A-270-ac-A","Index: 42<br />LibrarySize: 38432<br />is_blank: FALSE<br />rownames(df): BA4A-270B-H2O","Index: 43<br />LibrarySize: 38579<br />is_blank: FALSE<br />rownames(df): BA4A-270B-D2O-Form","Index: 44<br />LibrarySize: 38618<br />is_blank: FALSE<br />rownames(df): BA1B-250A-H2O","Index: 45<br />LibrarySize: 38770<br />is_blank: FALSE<br />rownames(df): BA4A-270A-D2O-Acet","Index: 46<br />LibrarySize: 39312<br />is_blank: FALSE<br />rownames(df): BA1D-120A-D2O-Form","Index: 47<br />LibrarySize: 39665<br />is_blank: FALSE<br />rownames(df): BA1B-250-bic-A","Index: 48<br />LibrarySize: 39750<br />is_blank: FALSE<br />rownames(df): BA4A-270A-D2O","Index: 49<br />LibrarySize: 39837<br />is_blank: FALSE<br />rownames(df): BA4A-270-ENV","Index: 50<br />LibrarySize: 40200<br />is_blank: FALSE<br />rownames(df): BA1B-250-ENV","Index: 51<br />LibrarySize: 41028<br />is_blank: FALSE<br />rownames(df): BA1B-250A-D2O-Acet","Index: 52<br />LibrarySize: 41253<br />is_blank: FALSE<br />rownames(df): BA4A-270A-D2O-Bic","Index: 53<br />LibrarySize: 41470<br />is_blank: FALSE<br />rownames(df): BA3A-270-ENV","Index: 54<br />LibrarySize: 41497<br />is_blank: FALSE<br />rownames(df): BA1D-120B-H2O","Index: 55<br />LibrarySize: 41559<br />is_blank: FALSE<br />rownames(df): BA4A-20-ac-A","Index: 56<br />LibrarySize: 41921<br />is_blank: FALSE<br />rownames(df): BA1B-250B-D2O-Bic","Index: 57<br />LibrarySize: 42889<br />is_blank: FALSE<br />rownames(df): BA1B-250A-D2O-Bic","Index: 58<br />LibrarySize: 44081<br />is_blank: FALSE<br />rownames(df): BA1B-150-ENV","Index: 59<br />LibrarySize: 44403<br />is_blank: FALSE<br />rownames(df): BA1D-120A-D2O","Index: 60<br />LibrarySize: 44949<br />is_blank: FALSE<br />rownames(df): BA4A-150-form-A","Index: 61<br />LibrarySize: 45475<br />is_blank: FALSE<br />rownames(df): BA1B-250A-D2O-Form","Index: 62<br />LibrarySize: 46017<br />is_blank: FALSE<br />rownames(df): BA4A-270A-H2O","Index: 63<br />LibrarySize: 46771<br />is_blank: FALSE<br />rownames(df): BA1D-120B-D2O","Index: 64<br />LibrarySize: 49213<br />is_blank: FALSE<br />rownames(df): BA4A-270-form-A","Index: 65<br />LibrarySize: 51047<br />is_blank: FALSE<br />rownames(df): BA4A-20-bic-A","Index: 66<br />LibrarySize: 59435<br />is_blank: FALSE<br />rownames(df): BA4A-150-ENV"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"FALSE","legendgroup":"FALSE","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[1,2,3,4,5,6,7,8,9,10,34],"y":[194,1003,1146,1278,1695,2504,2518,3259,6439,6722,36069],"text":["Index:  1<br />LibrarySize:   194<br />is_blank:  TRUE<br />rownames(df): KIT-BLANK-BOX2-3","Index:  2<br />LibrarySize:  1003<br />is_blank:  TRUE<br />rownames(df): KIT-BLANK-BOX3-1","Index:  3<br />LibrarySize:  1146<br />is_blank:  TRUE<br />rownames(df): NTC2","Index:  4<br />LibrarySize:  1278<br />is_blank:  TRUE<br />rownames(df): NTC1","Index:  5<br />LibrarySize:  1695<br />is_blank:  TRUE<br />rownames(df): SK-Filter-Ctrl-Blank2","Index:  6<br />LibrarySize:  2504<br />is_blank:  TRUE<br />rownames(df): OM23-Filter-Blank","Index:  7<br />LibrarySize:  2518<br />is_blank:  TRUE<br />rownames(df): SK-Filter-Ctrl-Blank1","Index:  8<br />LibrarySize:  3259<br />is_blank:  TRUE<br />rownames(df): KIT-BLANK_BOX2-2","Index:  9<br />LibrarySize:  6439<br />is_blank:  TRUE<br />rownames(df): KIT-BLANK-BOX1","Index: 10<br />LibrarySize:  6722<br />is_blank:  TRUE<br />rownames(df): KIT-BLANK-BOX2-1","Index: 34<br />LibrarySize: 36069<br />is_blank:  TRUE<br />rownames(df): TAC-Inc-Filter-Blank-A"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,196,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,191,196,1)"}},"hoveron":"points","name":"TRUE","legendgroup":"TRUE","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.958904109589042,"r":7.3059360730593621,"b":40.913242009132432,"l":54.794520547945211},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-2.25,69.25],"tickmode":"array","ticktext":["0","20","40","60"],"tickvals":[0,20,40,60],"categoryorder":"array","categoryarray":["0","20","40","60"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"Index","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-2768.0500000000002,62397.050000000003],"tickmode":"array","ticktext":["0","20000","40000","60000"],"tickvals":[0,20000,40000,60000],"categoryorder":"array","categoryarray":["0","20000","40000","60000"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"LibrarySize","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"is_blank","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"df5415dd0afd":{"x":{},"y":{},"colour":{},"label":{},"type":"scatter"}},"cur_data":"df5415dd0afd","visdat":{"df5415dd0afd":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>The library sizes of the positive samples range from low (hundreds of reads) to 60000 reads. There are some negative controls with plenty of reads and positive samples with few reads. There is one negative control “TAC-Inc-Filter-Blank-A” that corresponds to a blank filter that was used to assess contamination during filter sampling after incubation. However, most of these contaminants appear to be crossover from the other samples (ASVs 1-5). I do have a note in my notebook that I processed this blank <em>after</em> I cut out all of the other filters and removed them with tweezers: it is likely the tweezers were not sufficiently decontaminated with a simple ethanol wash. Luckily, the <code>decontam</code> package is sensitive to removing sample crossover. Overall, the blanks look really good. We shall proceed.</p>
</section>
<section id="identify-contaminants-prevalence" class="level3">
<h3 class="anchored" data-anchor-id="identify-contaminants-prevalence">Identify contaminants: prevalence</h3>
<p>The contamination id method we’ll use is the “prevalence” method. In this method, the prevalence (presence/absence across samples) of each sequence feature in true positive samples is compared to the prevalence in negative controls to identify contaminants.</p>
<p>In our phyloseq object, <code>is_blank</code> is the sample variable that holds the negative control information. We’ll summarize that data as a logical variable, with <code>TRUE</code> for controls, as this is the form required by the function <code>isContaminant</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>contamdf.prev <span class="ot">&lt;-</span> <span class="fu">isContaminant</span>(</span>
<span id="cb17-2"><a href="#cb17-2"></a>  ps, </span>
<span id="cb17-3"><a href="#cb17-3"></a>  <span class="at">method =</span> <span class="st">"prevalence"</span>,</span>
<span id="cb17-4"><a href="#cb17-4"></a>  <span class="at">neg =</span> <span class="st">"is_blank"</span>, </span>
<span id="cb17-5"><a href="#cb17-5"></a>  <span class="at">threshold =</span> <span class="fl">0.1</span></span>
<span id="cb17-6"><a href="#cb17-6"></a>  )</span>
<span id="cb17-7"><a href="#cb17-7"></a></span>
<span id="cb17-8"><a href="#cb17-8"></a><span class="fu">head</span>(contamdf.prev)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            freq prev p.freq    p.prev         p contaminant
ASV_1 0.41569402   64     NA 0.9871795 0.9871795       FALSE
ASV_2 0.16749942   63     NA 0.9981971 0.9981971       FALSE
ASV_3 0.04254611   63     NA 0.9981971 0.9981971       FALSE
ASV_4 0.04075008   60     NA 0.9998551 0.9998551       FALSE
ASV_5 0.03914110   60     NA 0.9998551 0.9998551       FALSE
ASV_6 0.01783033   57     NA 0.9999999 0.9999999       FALSE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="fu">table</span>(contamdf.prev<span class="sc">$</span>contaminant)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
  944    34 </code></pre>
</div>
</div>
<p>Prevalence-based contaminant identification has identified a number of contaminants, 34.</p>
<p>In the prevalence test there is a special value worth knowing, <code>threshold=0.5</code>, that will identify as contaminants all sequences thare are more prevalent in negative controls than in positive samples. Let’s try using this more aggressive classification threshold rather than the default. The default threshold for a contaminant is that it reaches a probability of <code>0.1</code> in the statistical test being performed.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>contamdf.prev05 <span class="ot">&lt;-</span> <span class="fu">isContaminant</span>(ps, <span class="at">method=</span><span class="st">"prevalence"</span>, <span class="at">neg=</span><span class="st">"is_blank"</span>, <span class="at">threshold=</span><span class="fl">0.5</span>)</span>
<span id="cb21-2"><a href="#cb21-2"></a><span class="fu">head</span>(contamdf.prev)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            freq prev p.freq    p.prev         p contaminant
ASV_1 0.41569402   64     NA 0.9871795 0.9871795       FALSE
ASV_2 0.16749942   63     NA 0.9981971 0.9981971       FALSE
ASV_3 0.04254611   63     NA 0.9981971 0.9981971       FALSE
ASV_4 0.04075008   60     NA 0.9998551 0.9998551       FALSE
ASV_5 0.03914110   60     NA 0.9998551 0.9998551       FALSE
ASV_6 0.01783033   57     NA 0.9999999 0.9999999       FALSE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="fu">table</span>(contamdf.prev05<span class="sc">$</span>contaminant)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
  894    84 </code></pre>
</div>
</div>
<p>Now there’s 84 contaminating ASVs.</p>
<p>Let’s take a look at the number of times several of these taxa were observed in negative controls and positive samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="co"># make phyloseq object of presence-absence in negative controls and true samples</span></span>
<span id="cb25-2"><a href="#cb25-2"></a>ps.pa <span class="ot">&lt;-</span> <span class="fu">transform_sample_counts</span>(ps, <span class="cf">function</span>(abund) <span class="dv">1</span><span class="sc">*</span>(abund<span class="sc">&gt;</span><span class="dv">0</span>))</span>
<span id="cb25-3"><a href="#cb25-3"></a>ps.pa.neg <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_data</span>(ps.pa)<span class="sc">$</span>is_blank <span class="sc">==</span> <span class="cn">TRUE</span>, ps.pa)</span>
<span id="cb25-4"><a href="#cb25-4"></a>ps.pa.pos <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(<span class="fu">sample_data</span>(ps.pa)<span class="sc">$</span>is_blank <span class="sc">==</span> <span class="cn">FALSE</span>, ps.pa)</span>
<span id="cb25-5"><a href="#cb25-5"></a></span>
<span id="cb25-6"><a href="#cb25-6"></a><span class="co"># make data frame of prevalence in positive and negative samples</span></span>
<span id="cb25-7"><a href="#cb25-7"></a><span class="co"># Make data.frame of prevalence in positive and negative samples</span></span>
<span id="cb25-8"><a href="#cb25-8"></a>df.pa <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pa.pos=</span><span class="fu">taxa_sums</span>(ps.pa.pos), <span class="at">pa.neg=</span><span class="fu">taxa_sums</span>(ps.pa.neg),</span>
<span id="cb25-9"><a href="#cb25-9"></a>                      <span class="at">contaminant=</span>contamdf.prev<span class="sc">$</span>contaminant)</span>
<span id="cb25-10"><a href="#cb25-10"></a><span class="fu">ggplot</span>(</span>
<span id="cb25-11"><a href="#cb25-11"></a>  <span class="at">data=</span>df.pa, </span>
<span id="cb25-12"><a href="#cb25-12"></a>  <span class="fu">aes</span>(<span class="at">x=</span>pa.neg, <span class="at">y=</span>pa.pos, <span class="at">color=</span>contaminant)) <span class="sc">+</span> </span>
<span id="cb25-13"><a href="#cb25-13"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb25-14"><a href="#cb25-14"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(contaminant)) <span class="sc">+</span></span>
<span id="cb25-15"><a href="#cb25-15"></a>  <span class="fu">xlab</span>(<span class="st">"Prevalence (Negative Controls)"</span>) <span class="sc">+</span> </span>
<span id="cb25-16"><a href="#cb25-16"></a>  <span class="fu">ylab</span>(<span class="st">"Prevalence (True Samples)"</span>) <span class="sc">+</span></span>
<span id="cb25-17"><a href="#cb25-17"></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>Samples seem to split pretty cleanly into a branch that shows up mostly in positive samples, and another that shows up mostly in negative controls, and the contaminant assignment (at default probability threshold) has done a good job of identifying those mostly in negative controls.</p>
<p>Here is our list of ASVs that are contaminants:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>contamdf.prev05 <span class="sc">|&gt;</span> <span class="fu">filter</span>(contaminant) <span class="sc">|&gt;</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              freq prev p.freq      p.prev           p contaminant
ASV_52 0.003626853   10     NA 0.031078490 0.031078490        TRUE
ASV_58 0.008831175   47     NA 0.062014556 0.062014556        TRUE
ASV_59 0.017330414   37     NA 0.005498439 0.005498439        TRUE
ASV_82 0.005641661    4     NA 0.326579670 0.326579670        TRUE
ASV_84 0.002625996    9     NA 0.002469421 0.002469421        TRUE
ASV_88 0.001207489    4     NA 0.006753663 0.006753663        TRUE</code></pre>
</div>
</div>
<p>We can make a list of these ASVs for filtering our phyloseq object:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>contams <span class="ot">&lt;-</span> contamdf.prev05 <span class="sc">|&gt;</span> </span>
<span id="cb28-2"><a href="#cb28-2"></a>  <span class="fu">filter</span>(contaminant <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> </span>
<span id="cb28-3"><a href="#cb28-3"></a>  <span class="fu">rownames</span>()</span>
<span id="cb28-4"><a href="#cb28-4"></a></span>
<span id="cb28-5"><a href="#cb28-5"></a>contams</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ASV_52"  "ASV_58"  "ASV_59"  "ASV_82"  "ASV_84"  "ASV_88"  "ASV_94" 
 [8] "ASV_97"  "ASV_102" "ASV_104" "ASV_106" "ASV_108" "ASV_118" "ASV_119"
[15] "ASV_120" "ASV_121" "ASV_123" "ASV_124" "ASV_141" "ASV_144" "ASV_145"
[22] "ASV_147" "ASV_149" "ASV_153" "ASV_163" "ASV_171" "ASV_172" "ASV_180"
[29] "ASV_193" "ASV_196" "ASV_200" "ASV_202" "ASV_210" "ASV_211" "ASV_217"
[36] "ASV_218" "ASV_222" "ASV_229" "ASV_235" "ASV_241" "ASV_261" "ASV_262"
[43] "ASV_271" "ASV_272" "ASV_284" "ASV_286" "ASV_300" "ASV_306" "ASV_308"
[50] "ASV_313" "ASV_319" "ASV_338" "ASV_344" "ASV_356" "ASV_362" "ASV_364"
[57] "ASV_370" "ASV_393" "ASV_396" "ASV_415" "ASV_427" "ASV_450" "ASV_463"
[64] "ASV_476" "ASV_477" "ASV_495" "ASV_501" "ASV_502" "ASV_521" "ASV_543"
[71] "ASV_566" "ASV_586" "ASV_615" "ASV_631" "ASV_633" "ASV_638" "ASV_677"
[78] "ASV_712" "ASV_721" "ASV_748" "ASV_799" "ASV_800" "ASV_812" "ASV_912"</code></pre>
</div>
</div>
<p>Similarly, we’ll make a list of non-contaminant taxa:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>not_contams <span class="ot">&lt;-</span> contamdf.prev05 <span class="sc">|&gt;</span> </span>
<span id="cb30-2"><a href="#cb30-2"></a>  <span class="fu">filter</span>(contaminant <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span> </span>
<span id="cb30-3"><a href="#cb30-3"></a>  <span class="fu">rownames</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="remove-contaminants-from-our-ps-object" class="level3">
<h3 class="anchored" data-anchor-id="remove-contaminants-from-our-ps-object">Remove contaminants from our PS object</h3>
<p>First, let’s use phyloseq’s <code>prune_taxa</code> to create a new ps of all our contaminants:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>ps.contaminants <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(ps, <span class="at">taxa =</span> contams)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we’ll do the same to create a fresh new PS object that is decontaminated:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a><span class="co"># define a list of samples to keep:</span></span>
<span id="cb32-2"><a href="#cb32-2"></a>pos_samples_list <span class="ot">&lt;-</span> metadata <span class="sc">|&gt;</span> </span>
<span id="cb32-3"><a href="#cb32-3"></a>  <span class="fu">filter</span>(<span class="sc">!</span>is_blank) <span class="sc">|&gt;</span> </span>
<span id="cb32-4"><a href="#cb32-4"></a>  <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span> </span>
<span id="cb32-5"><a href="#cb32-5"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb32-6"><a href="#cb32-6"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">rowname</span><span class="st">`</span>) <span class="sc">|&gt;</span> </span>
<span id="cb32-7"><a href="#cb32-7"></a>  <span class="fu">pull</span>()</span>
<span id="cb32-8"><a href="#cb32-8"></a></span>
<span id="cb32-9"><a href="#cb32-9"></a><span class="co"># this is the PS object we will carry forward with our analyses</span></span>
<span id="cb32-10"><a href="#cb32-10"></a>ps.decontam <span class="ot">&lt;-</span> ps <span class="sc">|&gt;</span> </span>
<span id="cb32-11"><a href="#cb32-11"></a>  <span class="co"># first we prune the taxa to select only non-contams:</span></span>
<span id="cb32-12"><a href="#cb32-12"></a>  <span class="fu">prune_taxa</span>(<span class="at">taxa =</span> not_contams) <span class="sc">|&gt;</span> </span>
<span id="cb32-13"><a href="#cb32-13"></a>  <span class="co"># then we remove the negative controls, as they have served their purpose:</span></span>
<span id="cb32-14"><a href="#cb32-14"></a>  <span class="fu">subset_samples</span>(is_blank <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span> </span>
<span id="cb32-15"><a href="#cb32-15"></a>  microViz<span class="sc">::</span><span class="fu">ps_mutate</span>(</span>
<span id="cb32-16"><a href="#cb32-16"></a>    <span class="at">depth_m =</span> <span class="fu">paste0</span>(depth, <span class="st">"m"</span>)</span>
<span id="cb32-17"><a href="#cb32-17"></a>  ) <span class="sc">|&gt;</span> </span>
<span id="cb32-18"><a href="#cb32-18"></a>  <span class="co"># validation step</span></span>
<span id="cb32-19"><a href="#cb32-19"></a>  microViz<span class="sc">::</span><span class="fu">tax_fix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, validate that the phyloseq object is in good shape:</p>
</section>
<section id="validate-and-finalize-our-ps-object" class="level3">
<h3 class="anchored" data-anchor-id="validate-and-finalize-our-ps-object">Validate and finalize our PS object</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="fu">phyloseq_validate</span>(ps.decontam, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 894 taxa and 55 samples ]
sample_data() Sample Data:       [ 55 samples by 6 sample variables ]
tax_table()   Taxonomy Table:    [ 894 taxa by 6 taxonomic ranks ]</code></pre>
</div>
</div>
</section>
</section>
<section id="cache" class="level2">
<h2 class="anchored" data-anchor-id="cache">Cache</h2>
<p>Write all of these important phyloseq objects to the repository’s cache, so they can be referenced later.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a><span class="co"># save original phyloseq object</span></span>
<span id="cb35-2"><a href="#cb35-2"></a><span class="fu">write_rds</span>(ps, <span class="st">"cache/ps.RDS"</span>)</span>
<span id="cb35-3"><a href="#cb35-3"></a></span>
<span id="cb35-4"><a href="#cb35-4"></a><span class="co"># save decontam'd PS object</span></span>
<span id="cb35-5"><a href="#cb35-5"></a><span class="fu">write_rds</span>(ps.decontam, <span class="st">"cache/ps-decontam.RDS"</span>)</span>
<span id="cb35-6"><a href="#cb35-6"></a></span>
<span id="cb35-7"><a href="#cb35-7"></a><span class="co"># save the contaminant PS object</span></span>
<span id="cb35-8"><a href="#cb35-8"></a><span class="fu">write_rds</span>(ps.contaminants, <span class="st">"cache/ps-contaminants.RDS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="analysis---all-samples" class="level1">
<h1>Analysis - All Samples</h1>
<p>First, let’s take a glance at the entire dataset to get a sense of what’s going on. This includes samples taken directly in the field, samples that TAC amended with <span class="math inline">\(^{13}C\)</span> substrates, and samples that SK amended with <span class="math inline">\(D_2O\)</span>.</p>
<section id="alpha-diversity-simpsons" class="level2">
<h2 class="anchored" data-anchor-id="alpha-diversity-simpsons">Alpha Diversity (Simpson’s)</h2>
<p>Alpha diversity describes “within-sample” diversity. It’s a measure of how diverse a single sample is, not taking into account observations of other samples.</p>
<p>Here we apply the Simpson index (or the “Gini-Simpson index”). The Simpson index is similar to the <a href="https://en.wikipedia.org/wiki/Diversity_index#Shannon_index">Shannon Index</a>: it is based on the probability that two entities (microbes, reads, etc.) taken from the sample at random are of different types (e.g., species). As this is a probability, the score ranges from 0 to 1.</p>
<p><span class="math display">\[
Simpson = 1 - \sum_{i = 1}^{R} p_i^2
\]</span></p>
<p>where <span class="math inline">\(p_i\)</span> is the proportion of observations (reads) belonging to the <span class="math inline">\(i\)</span>th species.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a><span class="fu">plot_richness</span>(ps.decontam, <span class="at">measures =</span> <span class="st">"Simpson"</span>, <span class="at">color =</span> <span class="st">"amendment"</span>, <span class="at">shape =</span> <span class="st">"well"</span>) <span class="sc">+</span></span>
<span id="cb36-2"><a href="#cb36-2"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb36-3"><a href="#cb36-3"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb36-4"><a href="#cb36-4"></a>       <span class="at">y =</span> <span class="st">"Simpson Index"</span>,</span>
<span id="cb36-5"><a href="#cb36-5"></a>       <span class="at">shape =</span> <span class="st">"Well"</span>,</span>
<span id="cb36-6"><a href="#cb36-6"></a>       <span class="at">color =</span> <span class="st">"Amendment"</span>) <span class="sc">+</span></span>
<span id="cb36-7"><a href="#cb36-7"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb36-8"><a href="#cb36-8"></a>  <span class="fu">theme</span>(</span>
<span id="cb36-9"><a href="#cb36-9"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">4</span>),</span>
<span id="cb36-10"><a href="#cb36-10"></a>    <span class="at">strip.text =</span> <span class="fu">element_blank</span>()</span>
<span id="cb36-11"><a href="#cb36-11"></a>    ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>Here, the thing that stands out most to me is greater alpha diversity (Simpson’s) in the “ENV” samples. Remember: these were the samples that were filtered on-site, and were not incubated with any carbon amendments.</p>
</section>
<section id="ordination" class="level2">
<h2 class="anchored" data-anchor-id="ordination">Ordination</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>ps.decontam.ord <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(ps.decontam, <span class="st">"NMDS"</span>, <span class="st">"bray"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Square root transformation
Wisconsin double standardization
Run 0 stress 0.143322 
Run 1 stress 0.1433194 
... New best solution
... Procrustes: rmse 0.002369973  max resid 0.01252273 
Run 2 stress 0.1433179 
... New best solution
... Procrustes: rmse 0.00134361  max resid 0.007556611 
... Similar to previous best
Run 3 stress 0.1433194 
... Procrustes: rmse 0.001342566  max resid 0.007552068 
... Similar to previous best
Run 4 stress 0.151485 
Run 5 stress 0.1433179 
... New best solution
... Procrustes: rmse 1.522611e-05  max resid 6.588221e-05 
... Similar to previous best
Run 6 stress 0.151485 
Run 7 stress 0.1433255 
... Procrustes: rmse 0.002294566  max resid 0.01238133 
Run 8 stress 0.1433194 
... Procrustes: rmse 0.001347177  max resid 0.007579669 
... Similar to previous best
Run 9 stress 0.1433187 
... Procrustes: rmse 0.001413809  max resid 0.007591529 
... Similar to previous best
Run 10 stress 0.1433248 
... Procrustes: rmse 0.002336829  max resid 0.01241407 
Run 11 stress 0.1433248 
... Procrustes: rmse 0.00233728  max resid 0.01241824 
Run 12 stress 0.1514889 
Run 13 stress 0.1514968 
Run 14 stress 0.1433255 
... Procrustes: rmse 0.002295859  max resid 0.01239511 
Run 15 stress 0.1433174 
... New best solution
... Procrustes: rmse 0.0004070757  max resid 0.002142019 
... Similar to previous best
Run 16 stress 0.1433187 
... Procrustes: rmse 0.001345978  max resid 0.007573656 
... Similar to previous best
Run 17 stress 0.1433337 
... Procrustes: rmse 0.002974948  max resid 0.01448661 
Run 18 stress 0.1433194 
... Procrustes: rmse 0.001399939  max resid 0.007569962 
... Similar to previous best
Run 19 stress 0.1433187 
... Procrustes: rmse 0.001346069  max resid 0.007574622 
... Similar to previous best
Run 20 stress 0.1515084 
*** Best solution repeated 4 times</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a><span class="fu">plot_ordination</span>(</span>
<span id="cb39-2"><a href="#cb39-2"></a>  ps.decontam,</span>
<span id="cb39-3"><a href="#cb39-3"></a>  ps.decontam.ord,</span>
<span id="cb39-4"><a href="#cb39-4"></a>  <span class="at">type=</span><span class="st">"samples"</span>,</span>
<span id="cb39-5"><a href="#cb39-5"></a>  <span class="at">color=</span> <span class="st">"amendment"</span>, </span>
<span id="cb39-6"><a href="#cb39-6"></a>  <span class="at">shape =</span> <span class="st">"experiment"</span>,</span>
<span id="cb39-7"><a href="#cb39-7"></a>  <span class="at">title=</span><span class="st">"taxa"</span></span>
<span id="cb39-8"><a href="#cb39-8"></a>  ) <span class="sc">+</span></span>
<span id="cb39-9"><a href="#cb39-9"></a>  <span class="co"># geom_polygon(aes(group = interaction(well, amendment, experiment),</span></span>
<span id="cb39-10"><a href="#cb39-10"></a>  <span class="co">#                  fill = amendment), alpha = 0.5) +</span></span>
<span id="cb39-11"><a href="#cb39-11"></a>  <span class="fu">labs</span>(</span>
<span id="cb39-12"><a href="#cb39-12"></a>    <span class="at">title =</span> <span class="st">""</span>,</span>
<span id="cb39-13"><a href="#cb39-13"></a>    <span class="at">shape =</span> <span class="st">"experiment"</span>,</span>
<span id="cb39-14"><a href="#cb39-14"></a>    <span class="at">color =</span> <span class="st">"Amendment"</span></span>
<span id="cb39-15"><a href="#cb39-15"></a>    ) <span class="sc">+</span></span>
<span id="cb39-16"><a href="#cb39-16"></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="1152"></p>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a>plotly<span class="sc">::</span><span class="fu">ggplotly</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-468fbb4f0f62d55fa827" style="width:100%;height:433px;"></div>
<script type="application/json" data-for="htmlwidget-468fbb4f0f62d55fa827">{"x":{"data":[{"x":[-0.34486136413860552,0.47252634347429762,-0.28223110567509047,-0.016306736666108868,0.18331106633065516,0.086038735497897895],"y":[-0.43251727959758679,-0.25913379915514395,-0.73178876904286616,0.25532369849408926,0.20325317615038321,0.22987192175338558],"text":["amendment: acetate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.344861364<br />NMDS2: -0.432517280","amendment: acetate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.472526343<br />NMDS2: -0.259133799","amendment: acetate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.282231106<br />NMDS2: -0.731788769","amendment: acetate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.016306737<br />NMDS2:  0.255323698","amendment: acetate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.183311066<br />NMDS2:  0.203253176","amendment: acetate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.086038735<br />NMDS2:  0.229871922"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"(acetate,13C-SIP)","legendgroup":"(acetate,13C-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.15573291793707428,0.033378297769774065,0.21535361256014726,0.44000340757759288],"y":[-0.2069095105682271,-0.19100665960547519,0.16706637163123078,0.41535339288364476],"text":["amendment: acetate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.155732918<br />NMDS2: -0.206909511","amendment: acetate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.033378298<br />NMDS2: -0.191006660","amendment: acetate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.215353613<br />NMDS2:  0.167066372","amendment: acetate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.440003408<br />NMDS2:  0.415353393"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"(acetate,D2O-SIP)","legendgroup":"(acetate,D2O-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.52003288701282235,0.59091032546083311,0.10245002246406708,0.27333177203944281,-0.0054052737094774581,0.2050374847335284],"y":[-0.56340163154073875,-0.34836553755491834,-0.65295492730204108,0.25407236635545288,0.15120807459382066,0.3444009301312545],"text":["amendment: bicarbonate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.520032887<br />NMDS2: -0.563401632","amendment: bicarbonate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.590910325<br />NMDS2: -0.348365538","amendment: bicarbonate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.102450022<br />NMDS2: -0.652954927","amendment: bicarbonate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.273331772<br />NMDS2:  0.254072366","amendment: bicarbonate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.005405274<br />NMDS2:  0.151208075","amendment: bicarbonate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.205037485<br />NMDS2:  0.344400930"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(163,165,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(163,165,0,1)"}},"hoveron":"points","name":"(bicarbonate,13C-SIP)","legendgroup":"(bicarbonate,13C-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.042722912342251257,0.0074342821147203647,0.31075034575722482,0.43777667292965683,0.40342523924834511,0.17821187304132236],"y":[-0.23451631831611003,-0.17030658604419002,-0.029138059453876101,-0.058883168420521284,0.21471095457720094,0.10296580932975366],"text":["amendment: bicarbonate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1: -0.042722912<br />NMDS2: -0.234516318","amendment: bicarbonate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.007434282<br />NMDS2: -0.170306586","amendment: bicarbonate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.310750346<br />NMDS2: -0.029138059","amendment: bicarbonate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.437776673<br />NMDS2: -0.058883168","amendment: bicarbonate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.403425239<br />NMDS2:  0.214710955","amendment: bicarbonate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.178211873<br />NMDS2:  0.102965809"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(163,165,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(163,165,0,1)"}},"hoveron":"points","name":"(bicarbonate,D2O-SIP)","legendgroup":"(bicarbonate,D2O-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.12370563977084546,-0.0092286445492356477,0.25496928607924751,0.40570130427722112,0.13042245109840087,0.37172481099914628],"y":[-0.27283902628934614,-0.36286110156035656,0.001870996219717265,-0.0089217867484655301,0.093053799126429068,0.1104998085990013],"text":["amendment: D2O<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1: -0.123705640<br />NMDS2: -0.272839026","amendment: D2O<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1: -0.009228645<br />NMDS2: -0.362861102","amendment: D2O<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.254969286<br />NMDS2:  0.001870996","amendment: D2O<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.405701304<br />NMDS2: -0.008921787","amendment: D2O<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.130422451<br />NMDS2:  0.093053799","amendment: D2O<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.371724811<br />NMDS2:  0.110499809"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,125,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,191,125,1)"}},"hoveron":"points","name":"(D2O,D2O-SIP)","legendgroup":"(D2O,D2O-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.41212646512814222,-0.25885008356244887,0.23548488157952824,0.075758753428377887,0.053711979527720791,-0.02314032205532492],"y":[-0.54240027227305887,-0.22832508620605196,-0.36752860436362744,0.19837544821228661,0.13181422629200026,0.44777399602687357],"text":["amendment: formate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.412126465<br />NMDS2: -0.542400272","amendment: formate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.258850084<br />NMDS2: -0.228325086","amendment: formate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.235484882<br />NMDS2: -0.367528604","amendment: formate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.075758753<br />NMDS2:  0.198375448","amendment: formate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1:  0.053711980<br />NMDS2:  0.131814226","amendment: formate<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.023140322<br />NMDS2:  0.447773996"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,176,246,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,176,246,1)"}},"hoveron":"points","name":"(formate,13C-SIP)","legendgroup":"(formate,13C-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.082478823175646018,0.028758567945044826,0.65676984860621657,0.50681896564876305,0.64636699127951747,0.43142628803908645],"y":[-0.32990114143631016,-0.3852910124387795,0.02325362799151116,0.007698605927899318,0.24852428855197209,0.26634771055278056],"text":["amendment: formate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.082478823<br />NMDS2: -0.329901141","amendment: formate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.028758568<br />NMDS2: -0.385291012","amendment: formate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.656769849<br />NMDS2:  0.023253628","amendment: formate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.506818966<br />NMDS2:  0.007698606","amendment: formate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.646366991<br />NMDS2:  0.248524289","amendment: formate<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.431426288<br />NMDS2:  0.266347711"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,176,246,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,176,246,1)"}},"hoveron":"points","name":"(formate,D2O-SIP)","legendgroup":"(formate,D2O-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-1.3284714618033062,-0.98752075542346052,-1.3722759605699357,-0.51273578372789319,-0.79059089158372831,-0.46251146409627775,-0.88628316223852222,-0.73717108717774571,-0.88929546148555672],"y":[-0.37661356385987693,-0.26400724496871081,-0.51429569154803856,0.86744720998151614,0.86843547991771675,0.82930569978040403,0.47164442527697453,0.46403021310958553,0.48553953281255313],"text":["amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -1.328471462<br />NMDS2: -0.376613564","amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.987520755<br />NMDS2: -0.264007245","amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -1.372275961<br />NMDS2: -0.514295692","amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.512735784<br />NMDS2:  0.867447210","amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.790590892<br />NMDS2:  0.868435480","amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.462511464<br />NMDS2:  0.829305700","amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.886283162<br />NMDS2:  0.471644425","amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.737171087<br />NMDS2:  0.464030213","amendment: pre-incubation<br />experiment: 13C-SIP<br />na.rm: TRUE<br />NMDS1: -0.889295461<br />NMDS2:  0.485539533"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(231,107,243,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(231,107,243,1)"}},"hoveron":"points","name":"(pre-incubation,13C-SIP)","legendgroup":"(pre-incubation,13C-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.21456523424881382,0.065914541170947819,0.61030266744635975,0.53681767624910859,0.29676276902804316,0.30503922395300848],"y":[-0.26006615155800628,-0.28524297272258653,0.014901992241252356,-0.057460205482874643,0.10181084580505384,0.16412150573204029],"text":["amendment: NA<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.214565234<br />NMDS2: -0.260066152","amendment: NA<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.065914541<br />NMDS2: -0.285242973","amendment: NA<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.610302667<br />NMDS2:  0.014901992","amendment: NA<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.536817676<br />NMDS2: -0.057460205","amendment: NA<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.296762769<br />NMDS2:  0.101810846","amendment: NA<br />experiment: D2O-SIP<br />na.rm: TRUE<br />NMDS1:  0.305039224<br />NMDS2:  0.164121506"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(127,127,127,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(127,127,127,1)"}},"hoveron":"points","name":"(NA,D2O-SIP)","legendgroup":"(NA,D2O-SIP)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.958904109589042,"r":7.3059360730593621,"b":40.913242009132432,"l":48.949771689497723},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1.4737282510287433,0.75822213906502423],"tickmode":"array","ticktext":["-1.0","-0.5","0.0","0.5"],"tickvals":[-1,-0.49999999999999989,0,0.5],"categoryorder":"array","categoryarray":["-1.0","-0.5","0.0","0.5"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"NMDS1","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.81179998149089527,0.94844669236574586],"tickmode":"array","ticktext":["-0.5","0.0","0.5"],"tickvals":[-0.5,0,0.50000000000000011],"categoryorder":"array","categoryarray":["-0.5","0.0","0.5"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"NMDS2","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"Amendment<br />experiment","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"df543342c56b":{"colour":{},"shape":{},"na.rm":{},"x":{},"y":{},"type":"scatter"}},"cur_data":"df543342c56b","visdat":{"df543342c56b":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Two things stand out here to me. First, the non-amended samples (what we find in the wells immediately upon sampling) are different from the amended samples. This is not surprising and can be boiled down to two effects:</p>
<ol type="1">
<li>Microbial community shift: we are taking microbes out of their native habitat and amending them with carbon sources. Note that the higher the carbon amendment, the further distance from the natural sample.</li>
<li>Sampling depth: on-site, we filtered anywhere from 300 - 600 mL of site fluid onto filters which were then sequenced. For the SIP incubation experiments, only ~50mL of incubated fluid was sequenced. This means that rare taxa are much more likely to be missed, which could cause depressions in alpha diversity (observed above) and differences in community “structure.”</li>
</ol>
<p>It is unclear whether these effects can be teased apart, so the most useful comparisons will be the intra-set comparisons. These sets would be:</p>
<ol type="1">
<li>Natural samples, filtered on site.</li>
<li>13C-amended samples</li>
<li>D2O-amended samples (with C stimulation)</li>
</ol>
<p>Now let’s examine how the wells cluster:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a><span class="fu">plot_ordination</span>(</span>
<span id="cb41-2"><a href="#cb41-2"></a>  ps.decontam,</span>
<span id="cb41-3"><a href="#cb41-3"></a>  ps.decontam.ord,</span>
<span id="cb41-4"><a href="#cb41-4"></a>  <span class="at">type=</span><span class="st">"samples"</span>,</span>
<span id="cb41-5"><a href="#cb41-5"></a>  <span class="at">color=</span> <span class="st">"amendment"</span>, </span>
<span id="cb41-6"><a href="#cb41-6"></a>  <span class="at">shape =</span> <span class="st">"well"</span>,</span>
<span id="cb41-7"><a href="#cb41-7"></a>  <span class="at">title=</span><span class="st">"taxa"</span></span>
<span id="cb41-8"><a href="#cb41-8"></a>  ) <span class="sc">+</span></span>
<span id="cb41-9"><a href="#cb41-9"></a>  <span class="co"># geom_polygon(aes(group = interaction(well, amendment, experiment),</span></span>
<span id="cb41-10"><a href="#cb41-10"></a>  <span class="co">#                  fill = amendment), alpha = 0.5) +</span></span>
<span id="cb41-11"><a href="#cb41-11"></a>  <span class="fu">labs</span>(</span>
<span id="cb41-12"><a href="#cb41-12"></a>    <span class="at">title =</span> <span class="st">""</span>,</span>
<span id="cb41-13"><a href="#cb41-13"></a>    <span class="at">shape =</span> <span class="st">"Well"</span>,</span>
<span id="cb41-14"><a href="#cb41-14"></a>    <span class="at">color =</span> <span class="st">"Amendment"</span></span>
<span id="cb41-15"><a href="#cb41-15"></a>    ) <span class="sc">+</span></span>
<span id="cb41-16"><a href="#cb41-16"></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="1152"></p>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a>plotly<span class="sc">::</span><span class="fu">ggplotly</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-c4a0b428a416487969e5" style="width:100%;height:433px;"></div>
<script type="application/json" data-for="htmlwidget-c4a0b428a416487969e5">{"x":{"data":[{"x":[-0.34486136413860552,0.47252634347429762,-0.28223110567509047,0.15573291793707428,0.033378297769774065],"y":[-0.43251727959758679,-0.25913379915514395,-0.73178876904286616,-0.2069095105682271,-0.19100665960547519],"text":["amendment: acetate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.344861364<br />NMDS2: -0.432517280","amendment: acetate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.472526343<br />NMDS2: -0.259133799","amendment: acetate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.282231106<br />NMDS2: -0.731788769","amendment: acetate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.155732918<br />NMDS2: -0.206909511","amendment: acetate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.033378298<br />NMDS2: -0.191006660"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"(acetate,BA1B)","legendgroup":"(acetate,BA1B)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.016306736666108868,0.18331106633065516,0.086038735497897895,0.21535361256014726,0.44000340757759288],"y":[0.25532369849408926,0.20325317615038321,0.22987192175338558,0.16706637163123078,0.41535339288364476],"text":["amendment: acetate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1: -0.016306737<br />NMDS2:  0.255323698","amendment: acetate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.183311066<br />NMDS2:  0.203253176","amendment: acetate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.086038735<br />NMDS2:  0.229871922","amendment: acetate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.215353613<br />NMDS2:  0.167066372","amendment: acetate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.440003408<br />NMDS2:  0.415353393"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(248,118,109,1)","opacity":1,"size":5.6692913385826778,"symbol":"cross-thin-open","line":{"width":1.8897637795275593,"color":"rgba(248,118,109,1)"}},"hoveron":"points","name":"(acetate,BA4A)","legendgroup":"(acetate,BA4A)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.52003288701282235,0.59091032546083311,0.10245002246406708,-0.042722912342251257,0.0074342821147203647],"y":[-0.56340163154073875,-0.34836553755491834,-0.65295492730204108,-0.23451631831611003,-0.17030658604419002],"text":["amendment: bicarbonate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.520032887<br />NMDS2: -0.563401632","amendment: bicarbonate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.590910325<br />NMDS2: -0.348365538","amendment: bicarbonate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.102450022<br />NMDS2: -0.652954927","amendment: bicarbonate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.042722912<br />NMDS2: -0.234516318","amendment: bicarbonate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.007434282<br />NMDS2: -0.170306586"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(163,165,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(163,165,0,1)"}},"hoveron":"points","name":"(bicarbonate,BA1B)","legendgroup":"(bicarbonate,BA1B)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.31075034575722482,0.43777667292965683],"y":[-0.029138059453876101,-0.058883168420521284],"text":["amendment: bicarbonate<br />well: BA1D<br />na.rm: TRUE<br />NMDS1:  0.310750346<br />NMDS2: -0.029138059","amendment: bicarbonate<br />well: BA1D<br />na.rm: TRUE<br />NMDS1:  0.437776673<br />NMDS2: -0.058883168"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(163,165,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(163,165,0,1)"}},"hoveron":"points","name":"(bicarbonate,BA1D)","legendgroup":"(bicarbonate,BA1D)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.27333177203944281,-0.0054052737094774581,0.2050374847335284,0.40342523924834511,0.17821187304132236],"y":[0.25407236635545288,0.15120807459382066,0.3444009301312545,0.21471095457720094,0.10296580932975366],"text":["amendment: bicarbonate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.273331772<br />NMDS2:  0.254072366","amendment: bicarbonate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1: -0.005405274<br />NMDS2:  0.151208075","amendment: bicarbonate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.205037485<br />NMDS2:  0.344400930","amendment: bicarbonate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.403425239<br />NMDS2:  0.214710955","amendment: bicarbonate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.178211873<br />NMDS2:  0.102965809"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(163,165,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"cross-thin-open","line":{"width":1.8897637795275593,"color":"rgba(163,165,0,1)"}},"hoveron":"points","name":"(bicarbonate,BA4A)","legendgroup":"(bicarbonate,BA4A)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.12370563977084546,-0.0092286445492356477],"y":[-0.27283902628934614,-0.36286110156035656],"text":["amendment: D2O<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.123705640<br />NMDS2: -0.272839026","amendment: D2O<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.009228645<br />NMDS2: -0.362861102"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,125,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,191,125,1)"}},"hoveron":"points","name":"(D2O,BA1B)","legendgroup":"(D2O,BA1B)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.25496928607924751,0.40570130427722112],"y":[0.001870996219717265,-0.0089217867484655301],"text":["amendment: D2O<br />well: BA1D<br />na.rm: TRUE<br />NMDS1:  0.254969286<br />NMDS2:  0.001870996","amendment: D2O<br />well: BA1D<br />na.rm: TRUE<br />NMDS1:  0.405701304<br />NMDS2: -0.008921787"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,125,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,191,125,1)"}},"hoveron":"points","name":"(D2O,BA1D)","legendgroup":"(D2O,BA1D)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.13042245109840087,0.37172481099914628],"y":[0.093053799126429068,0.1104998085990013],"text":["amendment: D2O<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.130422451<br />NMDS2:  0.093053799","amendment: D2O<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.371724811<br />NMDS2:  0.110499809"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,191,125,1)","opacity":1,"size":5.6692913385826778,"symbol":"cross-thin-open","line":{"width":1.8897637795275593,"color":"rgba(0,191,125,1)"}},"hoveron":"points","name":"(D2O,BA4A)","legendgroup":"(D2O,BA4A)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.41212646512814222,-0.25885008356244887,0.23548488157952824,0.082478823175646018,0.028758567945044826],"y":[-0.54240027227305887,-0.22832508620605196,-0.36752860436362744,-0.32990114143631016,-0.3852910124387795],"text":["amendment: formate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.412126465<br />NMDS2: -0.542400272","amendment: formate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.258850084<br />NMDS2: -0.228325086","amendment: formate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.235484882<br />NMDS2: -0.367528604","amendment: formate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.082478823<br />NMDS2: -0.329901141","amendment: formate<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.028758568<br />NMDS2: -0.385291012"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,176,246,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,176,246,1)"}},"hoveron":"points","name":"(formate,BA1B)","legendgroup":"(formate,BA1B)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.65676984860621657,0.50681896564876305],"y":[0.02325362799151116,0.007698605927899318],"text":["amendment: formate<br />well: BA1D<br />na.rm: TRUE<br />NMDS1:  0.656769849<br />NMDS2:  0.023253628","amendment: formate<br />well: BA1D<br />na.rm: TRUE<br />NMDS1:  0.506818966<br />NMDS2:  0.007698606"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,176,246,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(0,176,246,1)"}},"hoveron":"points","name":"(formate,BA1D)","legendgroup":"(formate,BA1D)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.075758753428377887,0.053711979527720791,-0.02314032205532492,0.64636699127951747,0.43142628803908645],"y":[0.19837544821228661,0.13181422629200026,0.44777399602687357,0.24852428855197209,0.26634771055278056],"text":["amendment: formate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.075758753<br />NMDS2:  0.198375448","amendment: formate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.053711980<br />NMDS2:  0.131814226","amendment: formate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1: -0.023140322<br />NMDS2:  0.447773996","amendment: formate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.646366991<br />NMDS2:  0.248524289","amendment: formate<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.431426288<br />NMDS2:  0.266347711"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,176,246,1)","opacity":1,"size":5.6692913385826778,"symbol":"cross-thin-open","line":{"width":1.8897637795275593,"color":"rgba(0,176,246,1)"}},"hoveron":"points","name":"(formate,BA4A)","legendgroup":"(formate,BA4A)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-1.3284714618033062,-0.98752075542346052,-1.3722759605699357],"y":[-0.37661356385987693,-0.26400724496871081,-0.51429569154803856],"text":["amendment: pre-incubation<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -1.328471462<br />NMDS2: -0.376613564","amendment: pre-incubation<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -0.987520755<br />NMDS2: -0.264007245","amendment: pre-incubation<br />well: BA1B<br />na.rm: TRUE<br />NMDS1: -1.372275961<br />NMDS2: -0.514295692"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(231,107,243,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(231,107,243,1)"}},"hoveron":"points","name":"(pre-incubation,BA1B)","legendgroup":"(pre-incubation,BA1B)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.51273578372789319,-0.79059089158372831,-0.46251146409627775],"y":[0.86744720998151614,0.86843547991771675,0.82930569978040403],"text":["amendment: pre-incubation<br />well: BA3A<br />na.rm: TRUE<br />NMDS1: -0.512735784<br />NMDS2:  0.867447210","amendment: pre-incubation<br />well: BA3A<br />na.rm: TRUE<br />NMDS1: -0.790590892<br />NMDS2:  0.868435480","amendment: pre-incubation<br />well: BA3A<br />na.rm: TRUE<br />NMDS1: -0.462511464<br />NMDS2:  0.829305700"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(231,107,243,1)","opacity":1,"size":5.6692913385826778,"symbol":"square","line":{"width":1.8897637795275593,"color":"rgba(231,107,243,1)"}},"hoveron":"points","name":"(pre-incubation,BA3A)","legendgroup":"(pre-incubation,BA3A)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.88628316223852222,-0.73717108717774571,-0.88929546148555672],"y":[0.47164442527697453,0.46403021310958553,0.48553953281255313],"text":["amendment: pre-incubation<br />well: BA4A<br />na.rm: TRUE<br />NMDS1: -0.886283162<br />NMDS2:  0.471644425","amendment: pre-incubation<br />well: BA4A<br />na.rm: TRUE<br />NMDS1: -0.737171087<br />NMDS2:  0.464030213","amendment: pre-incubation<br />well: BA4A<br />na.rm: TRUE<br />NMDS1: -0.889295461<br />NMDS2:  0.485539533"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(231,107,243,1)","opacity":1,"size":5.6692913385826778,"symbol":"cross-thin-open","line":{"width":1.8897637795275593,"color":"rgba(231,107,243,1)"}},"hoveron":"points","name":"(pre-incubation,BA4A)","legendgroup":"(pre-incubation,BA4A)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.21456523424881382,0.065914541170947819],"y":[-0.26006615155800628,-0.28524297272258653],"text":["amendment: NA<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.214565234<br />NMDS2: -0.260066152","amendment: NA<br />well: BA1B<br />na.rm: TRUE<br />NMDS1:  0.065914541<br />NMDS2: -0.285242973"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(127,127,127,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(127,127,127,1)"}},"hoveron":"points","name":"(NA,BA1B)","legendgroup":"(NA,BA1B)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.61030266744635975,0.53681767624910859],"y":[0.014901992241252356,-0.057460205482874643],"text":["amendment: NA<br />well: BA1D<br />na.rm: TRUE<br />NMDS1:  0.610302667<br />NMDS2:  0.014901992","amendment: NA<br />well: BA1D<br />na.rm: TRUE<br />NMDS1:  0.536817676<br />NMDS2: -0.057460205"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(127,127,127,1)","opacity":1,"size":5.6692913385826778,"symbol":"triangle-up","line":{"width":1.8897637795275593,"color":"rgba(127,127,127,1)"}},"hoveron":"points","name":"(NA,BA1D)","legendgroup":"(NA,BA1D)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.29676276902804316,0.30503922395300848],"y":[0.10181084580505384,0.16412150573204029],"text":["amendment: NA<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.296762769<br />NMDS2:  0.101810846","amendment: NA<br />well: BA4A<br />na.rm: TRUE<br />NMDS1:  0.305039224<br />NMDS2:  0.164121506"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(127,127,127,1)","opacity":1,"size":5.6692913385826778,"symbol":"cross-thin-open","line":{"width":1.8897637795275593,"color":"rgba(127,127,127,1)"}},"hoveron":"points","name":"(NA,BA4A)","legendgroup":"(NA,BA4A)","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":26.958904109589042,"r":7.3059360730593621,"b":40.913242009132432,"l":48.949771689497723},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1.4737282510287433,0.75822213906502423],"tickmode":"array","ticktext":["-1.0","-0.5","0.0","0.5"],"tickvals":[-1,-0.49999999999999989,0,0.5],"categoryorder":"array","categoryarray":["-1.0","-0.5","0.0","0.5"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"NMDS1","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.81179998149089527,0.94844669236574586],"tickmode":"array","ticktext":["-0.5","0.0","0.5"],"tickvals":[-0.5,0,0.50000000000000011],"categoryorder":"array","categoryarray":["-0.5","0.0","0.5"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"NMDS2","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"Well<br />Amendment","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"df54745f2a4c":{"colour":{},"shape":{},"na.rm":{},"x":{},"y":{},"type":"scatter"}},"cur_data":"df54745f2a4c","visdat":{"df54745f2a4c":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>As one might expect, the microbial communities of specific wells are similar to each other (see the pink clusters on the left). Microbial communities are more different between wells than they are between depths, suggesting that fluid depth is not the primary driver of microbial community composition. Furthermore, the amended samples cluster together by well. This suggests that microbial community shifts observed are specific to each well, <em>even taking into account variability introduced by different amendments</em>.</p>
</section>
</section>
<section id="analysis---1-field-samples" class="level1">
<h1>Analysis - (1) Field Samples</h1>
<p>Now that we’ve seen a high level overview, I’ll now focus on just the samples that were filtered on-site.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a>ps.om23.field <span class="ot">&lt;-</span> ps.decontam <span class="sc">|&gt;</span> </span>
<span id="cb43-2"><a href="#cb43-2"></a>  <span class="fu">subset_samples</span>(amendment <span class="sc">==</span> <span class="st">"pre-incubation"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb43-3"><a href="#cb43-3"></a>  <span class="co"># arrange our sample order by well, then depth</span></span>
<span id="cb43-4"><a href="#cb43-4"></a>  <span class="fu">ps_arrange</span>(well, depth)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="alpha-diversity-simpsons-1" class="level2">
<h2 class="anchored" data-anchor-id="alpha-diversity-simpsons-1">Alpha Diversity (Simpson’s)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a><span class="fu">plot_richness</span>(ps.om23.field, </span>
<span id="cb44-2"><a href="#cb44-2"></a>              <span class="at">measures =</span> <span class="st">"Simpson"</span>,</span>
<span id="cb44-3"><a href="#cb44-3"></a>              <span class="at">color =</span> <span class="st">"as.factor(depth)"</span>, </span>
<span id="cb44-4"><a href="#cb44-4"></a>              <span class="at">shape =</span> <span class="st">"well"</span></span>
<span id="cb44-5"><a href="#cb44-5"></a>              ) <span class="sc">+</span></span>
<span id="cb44-6"><a href="#cb44-6"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb44-7"><a href="#cb44-7"></a>  <span class="fu">scale_color_viridis_d</span>(<span class="at">option =</span> <span class="st">"plasma"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb44-8"><a href="#cb44-8"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb44-9"><a href="#cb44-9"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb44-10"><a href="#cb44-10"></a>       <span class="at">y =</span> <span class="st">"Simpson Index"</span>,</span>
<span id="cb44-11"><a href="#cb44-11"></a>       <span class="at">shape =</span> <span class="st">"Well"</span>,</span>
<span id="cb44-12"><a href="#cb44-12"></a>       <span class="at">color =</span> <span class="st">"Depth"</span>) <span class="sc">+</span></span>
<span id="cb44-13"><a href="#cb44-13"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb44-14"><a href="#cb44-14"></a>  <span class="fu">theme</span>(</span>
<span id="cb44-15"><a href="#cb44-15"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">4</span>),</span>
<span id="cb44-16"><a href="#cb44-16"></a>    <span class="at">strip.text =</span> <span class="fu">element_blank</span>()</span>
<span id="cb44-17"><a href="#cb44-17"></a>    ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>There isn’t much to pull out of this. Simpson index doesn’t change dramatically between the samples.</p>
</section>
<section id="composition" class="level2">
<h2 class="anchored" data-anchor-id="composition">Composition</h2>
<p>We’ll now calculate relative abundance of taxa in the natural samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a>ps.om23.field.relabund <span class="ot">&lt;-</span> ps.om23.field <span class="sc">|&gt;</span> </span>
<span id="cb45-2"><a href="#cb45-2"></a>  <span class="co"># use a lambda function to calculate relative abundance</span></span>
<span id="cb45-3"><a href="#cb45-3"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x)) <span class="sc">|&gt;</span> </span>
<span id="cb45-4"><a href="#cb45-4"></a>  <span class="co"># remove samples below 0.01 (1%) relative abundance</span></span>
<span id="cb45-5"><a href="#cb45-5"></a>  <span class="fu">filter_taxa</span>(<span class="cf">function</span>(x) <span class="fu">mean</span>(x) <span class="sc">&gt;</span> <span class="fl">0.01</span>, <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Bar chart time:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a>ps.om23.field.relabund <span class="sc">|&gt;</span> </span>
<span id="cb46-2"><a href="#cb46-2"></a>  <span class="co">#tax_agg("Genus") |&gt; </span></span>
<span id="cb46-3"><a href="#cb46-3"></a>  <span class="fu">comp_barplot</span>(<span class="at">tax_level =</span> <span class="st">"Genus"</span>,</span>
<span id="cb46-4"><a href="#cb46-4"></a>               <span class="at">sample_order =</span> <span class="fu">rev</span>(sample_list_ordered_field),</span>
<span id="cb46-5"><a href="#cb46-5"></a>               <span class="at">n_taxa =</span> <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb46-6"><a href="#cb46-6"></a>  <span class="fu">facet_grid</span>(</span>
<span id="cb46-7"><a href="#cb46-7"></a>    <span class="at">rows =</span> <span class="fu">vars</span>(well),</span>
<span id="cb46-8"><a href="#cb46-8"></a>    <span class="at">scales =</span> <span class="st">"free"</span>,</span>
<span id="cb46-9"><a href="#cb46-9"></a>    <span class="at">space =</span> <span class="st">"free"</span></span>
<span id="cb46-10"><a href="#cb46-10"></a>  ) <span class="sc">+</span></span>
<span id="cb46-11"><a href="#cb46-11"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb46-12"><a href="#cb46-12"></a>  <span class="fu">theme</span>(</span>
<span id="cb46-13"><a href="#cb46-13"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb46-14"><a href="#cb46-14"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb46-15"><a href="#cb46-15"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: otu_table of counts is NOT available!
otu_table of counts is NOT available!
Available otu_table contains 145 values that are not non-negative integers</code></pre>
</div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>Really interesting patterns here. Some observations.</p>
<ol type="1">
<li>All samples have significant populations of Methanobacterium.</li>
<li>BA1B has significant proportion of <em>Azospira.</em> These are nitrogen fixing bacteria.</li>
<li>BA3A has <em>Methanobacterium</em> abundances ranging between 40 - 50 %.</li>
<li>In BA3A especially (but also in BA1B-150), <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8698779/">D8A-2</a> is present at high abundance. This is an ulcutred class in the Firmicutes that was characterized to be involved in <a href="https://www.sciencedirect.com/science/article/pii/S0301479719304955?via%3Dihub#sec3">magnetite-associated degradation of VFAs</a> syntrophically coupled to methanogenesis.</li>
<li>BA4A has substantial sulfur cycling going on. First, we have Thermodesulfovibrionia class detected at moderate abundance. At high abundance is Sulfuritalea and SRB2. <a href="https://pubmed.ncbi.nlm.nih.gov/20709913/">Sulfuritalea</a> is a known hydrogenitrophic thiosulfate oxidizer. SRB2 is an uncharacterized sulfate reducing family that has been detected in <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2017.00308/full">CROMO</a>.</li>
<li>Meiothermus is detected in high abundance in BA4A, even at depth. Meiothermus was previously shown to <a href="https://link.springer.com/article/10.1134/s0026261723600234#Sec3">co-exist with Thermodesulfovibrionia SRB,</a> and their existence at depth was discussed by <a href="https://www.frontiersin.org/articles/10.3389/fmicb.2023.1138656/full#h5">Munro-Erlich et al.&nbsp;2023</a>.</li>
<li><a href="https://www.sciencedirect.com/topics/medicine-and-dentistry/magnetospirillum">Magnetospirillum</a>, a genus of magnetotactic bacteria, are detected in BA4A. COOL.</li>
<li>The sulfate reducer detected in BA3A is <em>not</em> Thermodesulfovibrio, but rather Desulfonatronum (and a bit of SRB2 Family).</li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/19921177/">Dechlorobacter</a> is detected in BA3A. This is a genus of perchlorate-reducing bacterium.</li>
</ol>
</section>
</section>
<section id="analysis---2-13c-experiments-tac" class="level1">
<h1>Analysis - (2) 13C Experiments (TAC)</h1>
<p>I’ll now subset all of the samples that correspond to <span class="math inline">\(^{13}C\)</span> experiments conducted by TAC.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>ps.decontam<span class="fl">.13</span>C <span class="ot">&lt;-</span> ps.decontam <span class="sc">|&gt;</span> </span>
<span id="cb48-2"><a href="#cb48-2"></a>  <span class="fu">subset_samples</span>(experiment <span class="sc">==</span> <span class="st">"13C-SIP"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="alpha-diversity-simpsons-2" class="level2">
<h2 class="anchored" data-anchor-id="alpha-diversity-simpsons-2">Alpha Diversity (Simpson’s)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a><span class="fu">plot_richness</span>(</span>
<span id="cb49-2"><a href="#cb49-2"></a>  ps.decontam<span class="fl">.13</span>C,</span>
<span id="cb49-3"><a href="#cb49-3"></a>  <span class="at">measures =</span> <span class="st">"Simpson"</span>, </span>
<span id="cb49-4"><a href="#cb49-4"></a>  <span class="at">color =</span> <span class="st">"as.factor(depth)"</span>, </span>
<span id="cb49-5"><a href="#cb49-5"></a>  <span class="at">shape =</span> <span class="st">"well"</span></span>
<span id="cb49-6"><a href="#cb49-6"></a>  ) <span class="sc">+</span></span>
<span id="cb49-7"><a href="#cb49-7"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(amendment)) <span class="sc">+</span></span>
<span id="cb49-8"><a href="#cb49-8"></a>  <span class="fu">scale_color_viridis_d</span>(<span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb49-9"><a href="#cb49-9"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb49-10"><a href="#cb49-10"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb49-11"><a href="#cb49-11"></a>       <span class="at">y =</span> <span class="st">"Simpson Index"</span>,</span>
<span id="cb49-12"><a href="#cb49-12"></a>       <span class="at">shape =</span> <span class="st">"Well"</span>,</span>
<span id="cb49-13"><a href="#cb49-13"></a>       <span class="at">color =</span> <span class="st">"Depth"</span>) <span class="sc">+</span></span>
<span id="cb49-14"><a href="#cb49-14"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb49-15"><a href="#cb49-15"></a>  <span class="fu">theme</span>(</span>
<span id="cb49-16"><a href="#cb49-16"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb49-17"><a href="#cb49-17"></a>    ) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>The same story as before. But I’ll add a bit more nuance:in BA1B, bicarbonate and acetate amendments exhibit decreased alpha diversity … but not formate.</p>
</section>
<section id="composition-1" class="level2">
<h2 class="anchored" data-anchor-id="composition-1">Composition</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a>ps.decontam<span class="fl">.13</span>C.relabund <span class="ot">&lt;-</span> ps.decontam<span class="fl">.13</span>C <span class="sc">|&gt;</span> </span>
<span id="cb50-2"><a href="#cb50-2"></a><span class="co"># use a lambda function to calculate relative abundance</span></span>
<span id="cb50-3"><a href="#cb50-3"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x)) </span>
<span id="cb50-4"><a href="#cb50-4"></a>  <span class="co"># remove samples below 0.01 (1%) relative abundance</span></span>
<span id="cb50-5"><a href="#cb50-5"></a>  <span class="co">#filter_taxa(function(x) mean(x) &gt; 0.01, TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>ps.decontam<span class="fl">.13</span>C.relabund <span class="sc">|&gt;</span> </span>
<span id="cb51-2"><a href="#cb51-2"></a>  <span class="co">#tax_agg("Genus") |&gt; </span></span>
<span id="cb51-3"><a href="#cb51-3"></a>  <span class="fu">comp_barplot</span>(<span class="at">tax_level =</span> <span class="st">"Genus"</span>,</span>
<span id="cb51-4"><a href="#cb51-4"></a>               <span class="at">n_taxa =</span> <span class="dv">30</span></span>
<span id="cb51-5"><a href="#cb51-5"></a>               ) <span class="sc">+</span></span>
<span id="cb51-6"><a href="#cb51-6"></a>  <span class="fu">facet_grid</span>(</span>
<span id="cb51-7"><a href="#cb51-7"></a>    <span class="at">rows =</span> <span class="fu">vars</span>(well, amendment),</span>
<span id="cb51-8"><a href="#cb51-8"></a>    <span class="at">scales =</span> <span class="st">"free"</span>,</span>
<span id="cb51-9"><a href="#cb51-9"></a>    <span class="at">space =</span> <span class="st">"free"</span></span>
<span id="cb51-10"><a href="#cb51-10"></a>  ) <span class="sc">+</span></span>
<span id="cb51-11"><a href="#cb51-11"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb51-12"><a href="#cb51-12"></a>  <span class="fu">theme</span>(</span>
<span id="cb51-13"><a href="#cb51-13"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb51-14"><a href="#cb51-14"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb51-15"><a href="#cb51-15"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: otu_table of counts is NOT available!
Available otu_table contains 3431 values that are not non-negative integers</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: otu_table of counts is NOT available!
Available otu_table contains 650 values that are not non-negative integers</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'seriation':
  method         from 
  reorder.hclust vegan</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: otu_table of counts is NOT available!
Available otu_table contains 650 values that are not non-negative integers</code></pre>
</div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>This seems to be evidence towards microbial community shifts. Methanobacterium and SRB2 relative abundance increases in the amended conditions. For example, in BA1B, SRB2 are barely detected, then they become significant community members.</p>
</section>
</section>
<section id="analysis---3-d2o-experiments" class="level1">
<h1>Analysis - (3) D2O Experiments</h1>
<p>Filter to the D2O samples:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a>ps.decontam.d2o <span class="ot">&lt;-</span> ps.decontam <span class="sc">|&gt;</span> </span>
<span id="cb56-2"><a href="#cb56-2"></a>  <span class="fu">subset_samples</span>(experiment <span class="sc">==</span> <span class="st">"D2O-SIP"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="composition-2" class="level2">
<h2 class="anchored" data-anchor-id="composition-2">Composition</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a>ps.decontam.d2o.relabund <span class="ot">&lt;-</span> ps.decontam.d2o <span class="sc">|&gt;</span> </span>
<span id="cb57-2"><a href="#cb57-2"></a><span class="co"># use a lambda function to calculate relative abundance</span></span>
<span id="cb57-3"><a href="#cb57-3"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x)) </span>
<span id="cb57-4"><a href="#cb57-4"></a>  <span class="co"># remove samples below 0.01 (1%) relative abundance</span></span>
<span id="cb57-5"><a href="#cb57-5"></a>  <span class="co">#filter_taxa(function(x) mean(x) &gt; 0.01, TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a>ps.decontam.d2o.relabund <span class="sc">|&gt;</span> </span>
<span id="cb58-2"><a href="#cb58-2"></a>  <span class="co">#tax_agg("Genus") |&gt; </span></span>
<span id="cb58-3"><a href="#cb58-3"></a>  <span class="fu">comp_barplot</span>(<span class="at">tax_level =</span> <span class="st">"Genus"</span>,</span>
<span id="cb58-4"><a href="#cb58-4"></a>               <span class="at">n_taxa =</span> <span class="dv">30</span></span>
<span id="cb58-5"><a href="#cb58-5"></a>               ) <span class="sc">+</span></span>
<span id="cb58-6"><a href="#cb58-6"></a>  <span class="fu">facet_grid</span>(</span>
<span id="cb58-7"><a href="#cb58-7"></a>    <span class="at">rows =</span> <span class="fu">vars</span>(well, amendment),</span>
<span id="cb58-8"><a href="#cb58-8"></a>    <span class="at">scales =</span> <span class="st">"free"</span>,</span>
<span id="cb58-9"><a href="#cb58-9"></a>    <span class="at">space =</span> <span class="st">"free"</span></span>
<span id="cb58-10"><a href="#cb58-10"></a>  ) <span class="sc">+</span></span>
<span id="cb58-11"><a href="#cb58-11"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb58-12"><a href="#cb58-12"></a>  <span class="fu">theme</span>(</span>
<span id="cb58-13"><a href="#cb58-13"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb58-14"><a href="#cb58-14"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb58-15"><a href="#cb58-15"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: otu_table of counts is NOT available!
Available otu_table contains 2281 values that are not non-negative integers</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: otu_table of counts is NOT available!
otu_table of counts is NOT available!
Available otu_table contains 660 values that are not non-negative integers</code></pre>
</div>
<div class="cell-output-display">
<p><img src="10_16S_Analysis_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>The community shift is more apparent: Methanobacterium and SRB2 combined make up &gt;60% of the microbial community in these incubations.</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>